app/views/memberships/index.html.erb

```diff
@@ -1,21 +1,25 @@
 <% @memberships.each do |membership| %>
-  <%= membership.name %>
-  <%= membership.email %>
-  <%= membership.status %>
-  <%= button_to "Approve", membership_approval_path(membership) %>
-  <% if ModeratorPolicy.new(current_person, membership.community).update? &&
-       membership.person %>
-    <% if membership.status != 'moderator' %>
-      <%= button_to "Set moderator",
-          community_moderator_path(membership.community, membership.person),
-          method: :put
-      %>
+  <div>
+    <%= membership.name %>
+    <%= membership.email %>
+    <%= membership.status %>
+    <% if membership.status == 'guest' && policy(membership).approval? %>
+      <%= button_to "Approve", membership_approval_path(membership) %>
     <% end %>
-    <% if membership.status == 'moderator' %>
-      <%= button_to "Dismiss moderator",
-          community_moderator_path(membership.community, membership.person),
-          method: :delete
-      %>
+    <% if ModeratorPolicy.new(current_person, membership.community).update? &&
+        membership.person %>
+      <% if membership.status != 'moderator' %>
+        <%= button_to "Set moderator",
+            community_moderator_path(membership.community, membership.person),
+            method: :put
+        %>
+      <% end %>
+      <% if membership.status == 'moderator' %>
+        <%= button_to "Dismiss moderator",
+            community_moderator_path(membership.community, membership.person),
+            method: :delete
+        %>
+      <% end %>
     <% end %>
-  <% end %>
+  </div>
```

Comment:
Not ideal UI but I planned to make the UI attractive and clean up these if-else in view when working on a UI improvement ticket. 
https://app.asana.com/0/1112086652417752/1119440732145137/f